packages/taxdat/R/postprocessing.R

```diff
@@ -0,0 +1,712 @@
+      dplyr::group_by(rid, x, y) %>% 
+      dplyr::slice(1) %>% 
+      dplyr::select(rid, x, y, geom)
+    
+    saveRDS(res, file = space_grid_sf_file)
+  }
+  
+  res
+}
+
+#' get_smooth_grid
+#'
+#' @param config_list 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+get_spacetime_grid <- function(config_list,
+                               redo = FALSE) {
+  
+  # Make file name
+  spacetime_grid_sf_file <- stringr::str_replace(config_list$file_names$observations_filename,
+                                                 ".preprocess.rdata",
+                                                 "_spacetime_grid_sf.rds")
+  
+  if (file.exists(spacetime_grid_sf_file) & !redo) {
+    res <- readRDS(spacetime_grid_sf_file)
+  } else {
+    res <- taxdat::read_file_of_type(config_list$file_names$stan_input_filename, "stan_input")$sf_grid
+    
+    saveRDS(res, file = spacetime_grid_sf_file)
+  }
+  
+  res
+}
+
+get_mean_pop_grid <- function(config_list,
+                              redo = FALSE) {
+  
+  # Make file name
+  mean_pop_grid_sf_file <- stringr::str_replace(config_list$file_names$observations_filename,
+                                                ".preprocess.rdata",
+                                                "_mean_pop_grid_sf.rds")
+  
+  if (file.exists(mean_pop_grid_sf_file) & !redo) {
+    res <- readRDS(mean_pop_grid_sf_file)
+  } else {
+    res <- get_spacetime_grid(config_list = config_list) %>% 
+      mutate(pop = taxdat::read_file_of_type(config_list$file_names$stan_input_filename, variable = "stan_input")$stan_data$pop) %>% 
+      sf::st_drop_geometry() %>% 
+      dplyr::group_by(rid, x, y, id) %>% 
+      dplyr::summarise(pop = mean(pop)) %>% 
+      dplyr::inner_join(get_space_grid(config_list = config_list), .)
+    
+    saveRDS(res, file = mean_pop_grid_sf_file)
+  }
+  
+  res
+}
+
+
+# Postprocess functions ---------------------------------------------------
+
+#' collapse_grid
+#' Function to collapse space grid to single non-overlapping cells
+#' 
+#' @param grid_obj 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+collapse_grid <- function(df) {
+  
+  u_grid <- df  %>%
+    group_by(rid, x, y) %>% 
+    slice(1) %>% 
+    select(rid, x, y, geom)
+  
+  
+  res <- df %>%
+    st_drop_geometry() %>% 
+    group_by(rid, x, y) %>% 
+    summarise(mean = mean(mean),
+              q5 = min(q5),
+              q95 = max(q95),
+              overlap = n() > 1) %>% 
+    inner_join(u_grid, .)
+  
+  res
+}
+
+
+# Auxilliary funcitons ----------------------------------------------------
+
+#' Title
+#' https://www.tutorialspoint.com/r/r_mean_median_mode.htm
+#' @param v 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+compute_mode <- function(v) {
+  uniqv <- unique(v)
+  uniqv[which.max(tabulate(match(v, uniqv)))]
+}
+
+#' Title
+#'
+#' @return
+#' @export
+#'
+#' @examples
+get_country_from_filename <- function(x) {
```

Comment:
Just noting that we also already have get_country_isocode as a function in covariate_helpers, which pulls the 3-letter ISO code from the config$countries argument. The 3-letter ISO code option is already enforced by the config settings. Not sure if this would be useful in the situation that this function is applied

Reply:
I don't understand the last sentence. Are you suggesting to remove this function since the iso code is already enforced in the config and use get_country_isocode instead? @eclee25

Reply:
Changed this to use get_country_isocode when possible, and changed get_country_from_filename to get_country_from_string to use in other settings.